src/PaymentGateways/PayPalCommerce/Webhooks/Listeners/PayPalCommerce/PaymentCaptureCompleted.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace Give\PaymentGateways\PayPalCommerce\Webhooks\Listeners\PayPalCommerce;
+
+/**
+ * Class PaymentCaptureCompleted
+ * @package Give\PaymentGateways\PayPalCommerce\Webhooks\Listeners\PayPalCommerce
+ *
+ * @since 2.8.0
+ */
+class PaymentCaptureCompleted extends PaymentEventListener {
+	/**
+	 * @inheritDoc
+	 */
+	public function processEvent( $event ) {
+		$paymentId = $this->getPaymentFromRefund( $event->resource, 'self' );
```

Comment:
@ravinderk I'm really confused by this. Why do you need to get the Payment from the API... again? By using the "self" link you're literally just grabbing the Payment _again_. In this context, `$event->resource` **is** a Payment, so all you need is `$event->resource->id`.

Reply:
fixed